docs/advanced_topics/integration_tests.md

```diff
@@ -0,0 +1,130 @@
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func \(s \*ApiTestSuite\) Test_Euro/ /\n}/)
+```go
+func (s *ApiTestSuite) Test_Euro() *suite.ApiServerTestCase {
+	return &suite.ApiServerTestCase{
+		Method:             http.MethodGet,
+		Url:                "/euro/10/GBP",
+		Headers:            map[string]string{},
+		ExpectedStatusCode: http.StatusOK,
+		Assert: func(response *resty.Response) error {
+			result, err := strconv.ParseFloat(string(response.Body()), 64)
+			s.NoError(err)
+			s.Equal(8.0, result)
+
+			return nil
+		},
+	}
+}
+```
+
+Notice the predefined structure of an _ApiServerTestCase_: it is an object that has fields for all the information needed in performing an HTTP call, the expected status code, and a method that tests the result for correctness.
+
+In the example from `examples/getting_started/integration/api_test.go` we can see that both types of test cases (_resty.Client_ and _ApiServerTestCase_) share the same `SetupSuite` method. Indeed, you can use all three types of tests in the same test suite.
+
+### Stream integration tests
+
+If you have a Gosoline application that takes its input from a stream, you need a test which can run your application locally, send data to a stream, and check your application's output for correctness. 
+
+The stream-consumer application, found in advanced/stream-consumer, reads unsigned integers from the inputEvent input, increments them by one, and publishes them to the publisher-outputEvent output. Below is an extract from its integration test:
+
+[embedmd]:# (../../examples/advanced/stream-consumer-test/stream_consumer_test.go /func \(s \*ConsumerTestSuite\) SetupSuite/ /\n}/)
+```go
+func (s *ConsumerTestSuite) SetupSuite() []suite.Option {
+	return []suite.Option{
+		suite.WithLogLevel("debug"),
+		suite.WithConfigFile("../stream-consumer/config.dist.yml"),
+		suite.WithModule("consumer", stream.NewConsumer("consumer", consumer.NewConsumer())),
+	}
+}
+```
+
+It is making use of the same `config.dist.yml` file as `stream-consumer`, and it will be using the module created by _consumer.NewConsumer_. The difference between a _StreamTestCase_ test and an _ApiServerTestCase_ can be seen in its test cases:
+
+[embedmd]:# (../../examples/advanced/stream-consumer-test/stream_consumer_test.go /func \(s \*ConsumerTestSuite\) TestSuccess/ /\n}/)
+```go
+func (s *ConsumerTestSuite) TestSuccess() *suite.StreamTestCase {
+	return &suite.StreamTestCase{
+		Input: map[string][]suite.StreamTestCaseInput{
+			"consumer": {
```

Comment:
Why is this called `consumer`? I know it works like that, but above you talked about the input. So how does gosoline know which input to write stuff to? And why do we specify the consumer here and not the input?